src/app/clusters/general-diagnostics-server/tests/TestGeneralDiagnosticsCluster.cpp

```diff
@@ -247,4 +352,133 @@ TEST_F(TestGeneralDiagnosticsCluster, AttributesTest)
     }
 }
 
+TEST_F(TestGeneralDiagnosticsCluster, TimeSnapshotCommandTest)
+{
+    // Create a cluster with no optional attributes enabled
+    const GeneralDiagnosticsCluster::OptionalAttributeSet optionalAttributeSet;
+    ScopedDiagnosticsProvider<NullProvider> nullProvider;
+    GeneralDiagnosticsCluster cluster(optionalAttributeSet);
+
+    // Prepare command invocation infrastructure
+    MockCommandHandler handler;
+    ConcreteCommandPath commandPath(kRootEndpointId, GeneralDiagnostics::Id, GeneralDiagnostics::Commands::TimeSnapshot::Id);
+    DataModel::InvokeRequest invokeRequest;
+    invokeRequest.path = commandPath;
+
+    uint8_t tlvBuffer[kEmptyCommandTLVBufferSize];
+    TLV::TLVReader reader;
+    SetupEmptyCommandTLV(tlvBuffer, reader);
+
+    // Invoke TimeSnapshot command
+    auto result = cluster.InvokeCommand(invokeRequest, reader, &handler);
+
+    // Verify that the command was handled successfully (returns nullopt on success)
+    EXPECT_FALSE(result.has_value());
+
+    // Verify that a response was generated
+    EXPECT_TRUE(handler.HasResponse());
+    EXPECT_EQ(handler.GetResponseCommandId(), GeneralDiagnostics::Commands::TimeSnapshotResponse::Id);
+
+    // Decode and verify the response
+    GeneralDiagnostics::Commands::TimeSnapshotResponse::DecodableType response;
+    ASSERT_EQ(handler.DecodeTimeSnapshotResponse(response), CHIP_NO_ERROR);
+
+    // Verify that systemTimeMs is present
+    EXPECT_GE(response.systemTimeMs, 0u);
+
+    // Basic configuration excludes POSIX time
+    EXPECT_TRUE(response.posixTimeMs.IsNull());
+}
+
+TEST_F(TestGeneralDiagnosticsCluster, TimeSnapshotCommandWithPosixTimeTest)
+{
+    // Configure cluster with POSIX time support enabled and no optional attributes enabled
+    const GeneralDiagnosticsCluster::OptionalAttributeSet optionalAttributeSet;
+    ScopedDiagnosticsProvider<NullProvider> nullProvider;
+    const GeneralDiagnosticsFunctionsConfig functionsConfig{
+        .enablePosixTime      = true,
+        .enablePayloadSnaphot = false,
+    };
+    GeneralDiagnosticsClusterFullConfigurable cluster(optionalAttributeSet, functionsConfig);
+
+    // Prepare command invocation infrastructure
+    MockCommandHandler handler;
+    ConcreteCommandPath commandPath(kRootEndpointId, GeneralDiagnostics::Id, GeneralDiagnostics::Commands::TimeSnapshot::Id);
+    DataModel::InvokeRequest invokeRequest;
+    invokeRequest.path = commandPath;
+
+    uint8_t tlvBuffer[kEmptyCommandTLVBufferSize];
```

Comment:
I think we should work on some infrastructure on writing these tests somehow cleaner/smaller. Allocating buffers and calling helpers seems painful.

We have some (but maybe not ideal) in https://github.com/project-chip/connectedhomeip/blob/master/src/app/data-model-provider/tests/ReadTesting.h for example and the same for Write. Maybe we should have something for commands.